crates/anvil/src/eth/otterscan/types.rs

```diff
@@ -231,59 +233,44 @@ impl OtsSearchTransactions {
     /// [`ots_getInternalOperations`](https://github.com/otterscan/otterscan/blob/develop/docs/custom-jsonrpc.md#ots_getinternaloperations)
-    pub fn batch_build(traces: Vec<Trace>) -> Vec<OtsInternalOperation> {
+    pub fn batch_build(traces: MinedTransaction) -> Vec<OtsInternalOperation> {
         traces
+            .info
+            .traces
+            .arena
             .iter()
-            .filter_map(|trace| {
-                match (trace.action.clone(), trace.result.clone()) {
-                    (Action::Call(Call { from, to, value, .. }), _) if !value.is_zero() => {
-                        Some(Self { r#type: OtsInternalOperationType::Transfer, from, to, value })
-                    }
-                    (
-                        Action::Create(Create { from, value, .. }),
-                        Some(Res::Create(CreateResult { address, .. })),
-                    ) => Some(Self {
-                        r#type: OtsInternalOperationType::Create,
-                        from,
-                        to: address,
-                        value,
-                    }),
-                    (Action::Suicide(Suicide { address, .. }), _) => {
-                        // this assumes a suicide trace always has a parent trace
-                        let (from, value) =
-                            Self::find_suicide_caller(&traces, &trace.trace_address).unwrap();
-
-                        Some(Self {
-                            r#type: OtsInternalOperationType::SelfDestruct,
-                            from,
-                            to: address,
-                            value,
-                        })
-                    }
-                    _ => None,
+            .filter_map(|node| match (node.kind(), node.status()) {
+                (CallKind::Call, _) if !node.trace.value.is_zero() => Some(Self {
+                    r#type: OtsInternalOperationType::Transfer,
+                    from: node.trace.caller,
+                    to: node.trace.address,
+                    value: node.trace.value,
+                }),
+                (CallKind::Create, _) => Some(Self {
+                    r#type: OtsInternalOperationType::Create,
+                    from: node.trace.caller,
+                    to: node.trace.address,
+                    value: node.trace.value,
+                }),
+                (CallKind::Create2, _) => Some(Self {
+                    r#type: OtsInternalOperationType::Create2,
+                    from: node.trace.caller,
+                    to: node.trace.address,
+                    value: node.trace.value,
+                }),
+                (_, InstructionResult::SelfDestruct) => {
+                    Some(Self {
+                        r#type: OtsInternalOperationType::SelfDestruct,
+                        from: node.trace.address,
+                        // the foundry CallTraceNode doesn't have a refund address
```

Comment:
this is a useful comment, this will be supported soon

Reply:
That's good to know, if there's any way I can contribute or assist with this, i would be up for it !